### PR TITLE
CLI improvements ROUND 2

### PR DESCRIPTION
**Pull request by jquense**
Hopefully with 100% less broken things

### CODE REVIEW

**Comment by KyleAMathews** [comment]
Deploy preview failed.

Built with commit 5fd9a5d1714ad64c2d105dd1181935c65ac6c48e

https://app.netlify.com/sites/using-glamor/deploys/59a6daed0752d049cb5aac0f

**Comment by jquense** [comment]
ping :P

**Comment by KyleAMathews** [comment]
Oh I was waiting for you to tell me it was ready :-)

I'll give it a quick review.

It has a couple conflicting files.

File: www/package.json [hunk 1]
```diff
@@ -5,7 +5,7 @@
   "author": "Kyle Mathews <mathews.kyle@gmail.com>",
   "dependencies": {
     "bluebird": "^3.4.6",
-    "gatsby": "^1.9.10",
+    "gatsby": "file:../packages/gatsby",
```
Comment (KyleAMathews) [inline www/package.json]: You should use gatsby-dev-cli for copying over files during development

File: packages/gatsby/src/utils/reporter/errors.js [hunk 2]
```diff
@@ -0,0 +1,67 @@
+// @flow
```
Comment (KyleAMathews) [inline packages/gatsby/src/utils/reporter/errors.js]: This should be in gatsby-cli? Or is this a duplicate?

File: www/package.json (hunk 1 above)
Comment (jquense) [reply www/package.json]: yeah normally i do, but in cases where node_modules changed this ends up being easier...i wish yarn didn't save this tho.

File: packages/gatsby/src/utils/reporter/errors.js (hunk 2 above)
Comment (jquense) [reply packages/gatsby/src/utils/reporter/errors.js]: DUP!

File: packages/gatsby-cli/src/create-cli.js [hunk 3]
```diff
@@ -0,0 +1,194 @@
...
+  }
+
+  function getCommandHandler(command, handler) {
+    return argv => {
+      report.setVerbose(!!argv.verbose)
+
+      process.env.gatsby_log_level = argv.verbose ? `verbose` : `normal`
+      report.verbose(`set gatsby_log_level: "${process.env.gatsby_log_level}"`)
+
+      process.env.gatsby_executing_command === command
```
Comment (jquense) [inline packages/gatsby-cli/src/create-cli.js]: @KyleAMathews i added these just now, similar to how npm does I think this is the easiest way to handle checks for this across package instances and the like. WDYT?

**Comment by jquense** [comment]
tested and seemed to work well for me :)

File: packages/gatsby-plugin-netlify/create-redirects.js [hunk 4]
```diff
@@ -0,0 +1,90 @@
+"use strict";
```
Comment (KyleAMathews) [inline packages/gatsby-plugin-netlify/create-redirects.js]: more .gitignore fail

File: packages/gatsby-source-lever/normalize.js [hunk 5]
```diff
@@ -0,0 +1,214 @@
+"use strict";
```
Comment (KyleAMathews) [inline packages/gatsby-source-lever/normalize.js]: .gitignore fail

File: packages/gatsby-cli/src/create-cli.js (hunk 3 above)
Comment (KyleAMathews) [reply packages/gatsby-cli/src/create-cli.js]: Did you mean do an equality check here? Maybe not sure what you're doing :-)

File: packages/gatsby-source-lever/normalize.js (hunk 5 above)
Comment (jquense) [reply packages/gatsby-source-lever/normalize.js]: i'mm go through and try and address all these in a PR

File: packages/gatsby-cli/src/create-cli.js (hunk 3 above)
Comment (jquense) [reply packages/gatsby-cli/src/create-cli.js]: lol oops

**Comment by jquense** [comment]
should be gtg

**Comment by KyleAMathews** [comment]
Sweet! This published to canary?

**Comment by jquense** [comment]
not this most recent push

**Comment by KyleAMathews** [comment]
🎉 

Thanks @jquense! This is an awesome refactor with some really nice UX improvements to boot!

**Comment by jquense** [comment]
woohoo!!

**Pull request merged**